### ISSUE TITLE
[Bug]: [kafka][nightly] Test case hangs when using kafka as mq

### ISSUE THREAD

**Issue by binbinlv** [milvus-io/milvus#16499]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: latest
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2):pymilvus 2.0.2.dev10
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Test case hangs when using kafka as mq

### Expected Behavior

Test cases are completed when using kafka as mq

### Steps To Reproduce

```markdown
Run nightly:

https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-nightly-ci/detail/master/462/pipeline/141

&&

https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-nightly-ci/detail/master/461/pipeline/141


```

**Comment by binbinlv** [comment]
Similar issue occurs in the following nighlty:
https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-nightly-ci/detail/master/478/pipeline/136

**Comment by binbinlv** [comment]
Similar issue occurs in the following nightly:
(1) https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-nightly-ci/detail/master/483/pipeline

(2) https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-nightly-ci/detail/master/482/pipeline

**Comment by binbinlv** [comment]
This issue reproduced in high probability recently:
1. https://ci.milvus.io:18080/jenkins/job/milvus-nightly-ci/job/master/488/
(1) milvus logs:
[artifacts-milvus-distributed-kafka-master-488-pymilvus-e2e-logs.tar.gz](https://ci.milvus.io:18080/jenkins/job/milvus-nightly-ci/job/master/488/artifact/artifacts-milvus-distributed-kafka-master-488-pymilvus-e2e-logs.tar.gz)

(2) test logs:
https://ci.milvus.io:18080/jenkins/blue/rest/organizations/jenkins/pipelines/milvus-nightly-ci/branches/master/runs/488/nodes/64/steps/178/log/?start=0

2. https://ci.milvus.io:18080/jenkins/job/milvus-nightly-ci/job/master/487/
(1) milvus logs:
[artifacts-milvus-distributed-kafka-master-487-pymilvus-e2e-logs.tar.gz](https://ci.milvus.io:18080/jenkins/job/milvus-nightly-ci/job/master/487/artifact/artifacts-milvus-distributed-kafka-master-487-pymilvus-e2e-logs.tar.gz)

(2) test logs:
https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-nightly-ci/detail/master/487/pipeline/141

**Comment by jaime0815** [comment]
> This issue has been reproduced in high probability recently:
> 
> 1. https://ci.milvus.io:18080/jenkins/job/milvus-nightly-ci/job/master/488/
>    (1) milvus logs:
>    [artifacts-milvus-distributed-kafka-master-488-pymilvus-e2e-logs.tar.gz](https://ci.milvus.io:18080/jenkins/job/milvus-nightly-ci/job/master/488/artifact/artifacts-milvus-distributed-kafka-master-488-pymilvus-e2e-logs.tar.gz)
>    (2) test logs:
>    https://ci.milvus.io:18080/jenkins/blue/rest/organizations/jenkins/pipelines/milvus-nightly-ci/branches/master/runs/488/nodes/64/steps/178/log/?start=0
> 2. https://ci.milvus.io:18080/jenkins/job/milvus-nightly-ci/job/master/487/
>    (1) milvus logs:
>    [artifacts-milvus-distributed-kafka-master-487-pymilvus-e2e-logs.tar.gz](https://ci.milvus.io:18080/jenkins/job/milvus-nightly-ci/job/master/487/artifact/artifacts-milvus-distributed-kafka-master-487-pymilvus-e2e-logs.tar.gz)
>    (2) test logs:
>    https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-nightly-ci/detail/master/487/pipeline/141

487 pipeline hang caused by datanode core dump

```
[2022/04/23 15:04:05.103 +00:00] [INFO] [flow_graph_dd_node.go:286] ["ddNode add flushed segment"] [channelName=by-dev-rootcoord-dml_115_432727937879113731v19] [pChannelName=by-dev-rootcoord-dml_115]

panic: interface conversion: kafka.Event is nil, not *kafka.Message



goroutine 2242 [running]:

github.com/milvus-io/milvus/internal/mq/msgstream/mqwrapper/kafka.(*kafkaProducer).Send(0xc000fbe240, 0x0, 0x0, 0xc000678400, 0xc00174ace8, 0x3a13280, 0xc000d20c90, 0x3f67780)

        /go/src/github.com/milvus-io/milvus/internal/mq/msgstream/mqwrapper/kafka/kafka_producer.go:37 +0x1ed

github.com/milvus-io/milvus/internal/mq/msgstream.(*mqMsgStream).Produce(0xc001724d80, 0xc000fd3ec0, 0x0, 0x0)

        /go/src/github.com/milvus-io/milvus/internal/mq/msgstream/mq_msgstream.go:284 +0x4a9

github.com/milvus-io/milvus/internal/datanode.(*ddNode).sendDeltaTimeTick(0xc000e59a40, 0x6015bce44700002, 0x78, 0x90)

        /go/src/github.com/milvus-io/milvus/internal/datanode/flow_graph_dd_node.go:255 +0x1f8

github.com/milvus-io/milvus/internal/datanode.(*ddNode).forwardDeleteMsg(0xc000e59a40, 0x0, 0x0, 0x0, 0x0, 0x6015bce44700002, 0x1, 0x2)

        /go/src/github.com/milvus-io/milvus/internal/datanode/flow_graph_dd_node.go:228 +0x5a

github.com/milvus-io/milvus/internal/datanode.(*ddNode).Operate(0xc000e59a40, 0xc000ada400, 0x1, 0x1, 0xc00077e000, 0xc0006e2cc0, 0x57)

        /go/src/github.com/milvus-io/milvus/internal/datanode/flow_graph_dd_node.go:166 +0x52d

github.com/milvus-io/milvus/internal/util/flowgraph.(*nodeCtx).work(0xc000201dc0)

        /go/src/github.com/milvus-io/milvus/internal/util/flowgraph/node.go:100 +0x214

created by github.com/milvus-io/milvus/internal/util/flowgraph.(*nodeCtx).Start

        /go/src/github.com/milvus-io/milvus/internal/util/flowgraph/node.go:68 +0x55
```

**Comment by binbinlv** [comment]
Reproduced again:
https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-nightly-ci/detail/master/498/pipeline/145

milvus logs:
[artifacts-milvus-distributed-kafka-master-498-pymilvus-e2e-logs.tar.gz](https://ci.milvus.io:18080/jenkins/job/milvus-nightly-ci/job/master/498/artifact/artifacts-milvus-distributed-kafka-master-498-pymilvus-e2e-logs.tar.gz)

**Comment by binbinlv** [comment]
@jaime0815 
Similar phenomenon in the latest nightly, could you please have a check?

1. logs:
[artifacts-milvus-distributed-kafka-master-520-pymilvus-e2e-logs.tar.gz](https://ci.milvus.io:18080/jenkins/job/milvus-nightly-ci/job/master/520/artifact/artifacts-milvus-distributed-kafka-master-520-pymilvus-e2e-logs.tar.gz)

2. test pipeline logs:
https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-nightly-ci/detail/master/520/pipeline

**Comment by jaime0815** [comment]
> @jaime0815 Similar phenomenon in the latest nightly, could you please have a check?
> 
> 1. logs:
>    [artifacts-milvus-distributed-kafka-master-520-pymilvus-e2e-logs.tar.gz](https://ci.milvus.io:18080/jenkins/job/milvus-nightly-ci/job/master/520/artifact/artifacts-milvus-distributed-kafka-master-520-pymilvus-e2e-logs.tar.gz)
> 2. test pipeline logs:
>    https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-nightly-ci/detail/master/520/pipeline

release collection hang

**Comment by jaime0815** [comment]
> > @jaime0815 Similar phenomenon in the latest nightly, could you please have a check?
> > 
> > 1. logs:
> >    [artifacts-milvus-distributed-kafka-master-520-pymilvus-e2e-logs.tar.gz](https://ci.milvus.io:18080/jenkins/job/milvus-nightly-ci/job/master/520/artifact/artifacts-milvus-distributed-kafka-master-520-pymilvus-e2e-logs.tar.gz)
> > 2. test pipeline logs:
> >    https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-nightly-ci/detail/master/520/pipeline
> 
> release collection hang

fixed  on #16879

**Comment by binbinlv** [comment]
Did not hang last nightly, so close this issue first, and will reopen it if it occurs in the future.
